From c1b8a69774326b4fd23d06746ae7e6d937f9853c Mon Sep 17 00:00:00 2001
From: Vihang Karajgaonkar <vihang@cloudera.com>
Date: Wed, 31 Aug 2016 12:28:05 -0700
Subject: [PATCH 0747/1164] CLOUDERA-BUILD : Fix for test failure
 inputwherefalse.q

The fix for HIVE-14614 works in upstream but doesn't work in CDH due to differences in the configuration values for upstream tests and CDH as of this change. This fix is needed so that inputwherefalse.q test passes. The backport for the HIVE-14614 was reverted earlier and this standalone fix is needed to fix the test failures

Change-Id: Icadfdccc0bdafca58ce7adfb3907a20c849164a3
---
 .../hadoop/hive/ql/parse/SemanticAnalyzer.java     |    7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/ql/src/java/org/apache/hadoop/hive/ql/parse/SemanticAnalyzer.java b/ql/src/java/org/apache/hadoop/hive/ql/parse/SemanticAnalyzer.java
index 71fa248..c18ce2e 100644
--- a/ql/src/java/org/apache/hadoop/hive/ql/parse/SemanticAnalyzer.java
+++ b/ql/src/java/org/apache/hadoop/hive/ql/parse/SemanticAnalyzer.java
@@ -6439,7 +6439,12 @@ private Operator genFileSinkPlan(String dest, QB qb, Operator input)
     // it should be the same as the MoveWork's sourceDir.
     fileSinkDesc.setStatsAggPrefix(fileSinkDesc.getDirName().toString());
     if (HiveConf.getVar(conf, HIVESTATSDBCLASS).equalsIgnoreCase(StatDB.fs.name())) {
-      String statsTmpLoc = ctx.getTempDirForPath(dest_path).toString();
+      String statsTmpLoc;
+      if(isLocal) {
+        statsTmpLoc = ctx.getTempDirForPath(queryTmpdir).toString();
+      } else {
+        statsTmpLoc = ctx.getTempDirForPath(dest_path).toString();
+      }
       LOG.info("Set stats collection dir : " + statsTmpLoc);
       conf.set(StatsSetupConst.STATS_TMP_LOC, statsTmpLoc);
     }
-- 
1.7.9.5

